core-plugins/src/main/java/io/cdap/plugin/batch/joiner/JoinerConfig.java

```diff
@@ -108,11 +121,30 @@ public String getRequiredInputs() {
     return requiredInputs;
   }
 
+  @Nullable
+  public Schema getOutputSchema(FailureCollector collector) {
+    try {
+      return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      collector.addFailure("Invalid schema: " + e.getMessage(), null).withConfigProperty(OUTPUT_SCHEMA);
+    }
+    // if there was an error that was added, it will throw an exception, otherwise, this statement will not be executed
+    throw collector.getOrThrowException();
+  }
+
+  public boolean inputSchemasAvailable(Map<String, Schema> inputSchemas) {
+    // TODO: Remove isEmpty() check when CDAP-16351 is fixed
+    return !inputSchemas.isEmpty() && !inputSchemas.values().stream().anyMatch(v -> v == null);
```

Comment:
better: `return !inputSchemas.isEmpty() && inputSchemas.values().stream().noneMatch(Objects::isNull);`

Reply:
the method can be package private

Reply:
fixed.